test/e2e/suites/e2e/e2e_test.go

```diff
@@ -1067,3 +1161,16 @@ func isCloudProviderInitialized(taints []corev1.Taint) bool {
 	}
 	return true
 }
+
+func getServerIDFromMachineName(ctx context.Context, c crclient.Client, namespace string, machineName string) (string, error) {
+	machine := &infrav1.OpenStackMachine{}
+	key := crclient.ObjectKey{
+		Namespace: namespace,
+		Name:      machineName,
+	}
+	if err := c.Get(ctx, key, machine); err != nil {
+		return "", err
+	}
+
+	return *machine.Status.InstanceID, nil
```

Comment:
from the logs, it looks like this might be nil. Could it be that the machine was not ready yet?

can you add a check that verifies if it's nil? and if so, it waits for the machine to be ready?

Reply:
another option might be to keep checking the machine to see if it was deleted, instead of checking if the server was deleted this way we can abstract when server will be created/deleted.

Reply:
Thanks @MaysaMacedo. We've already got the ports and the trunk, so I think the machine should have been there :-? But I'll check that.

Reply:
It seems to me that somehow `machine.Status.InstanceID` never got updated, but both the machine and the server were ready. I changed the logic used for fetching the server id. The test passed now xD